# Profitable App Profiles Google Play Store

Our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.
This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

In [86]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [87]:
Android_app_raw = pd.read_csv('./data/googleplaystore.csv')

In [88]:
# Google Play store
Android_app_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


The Anroid_app_raw Dataframe has **10841** apps and 13 columns.

The columns that might be useful for the purpose of our analysis are `App`, `Category`, `Reviews`, `Installs`, `Type`, `Price`, `Genres`

# Google Play Store

In [89]:
# load data
Android_app_raw.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [178]:
# Remove irrelevant columns
Android_app = Android_app_raw[['App', 'Category', 'Reviews', 'Rating', 'Installs', 'Type', 'Price', 'Genres']]
Android_app.head()

App        Category Reviews  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     159   
1                                Coloring book moana  ART_AND_DESIGN     967   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN   87510   
3                              Sketch - Draw & Paint  ART_AND_DESIGN  215644   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     967   

   Rating     Installs  Type Price                     Genres  
0     4.1      10,000+  Free     0               Art & Design  
1     3.9     500,000+  Free     0  Art & Design;Pretend Play  
2     4.7   5,000,000+  Free     0               Art & Design  
3     4.5  50,000,000+  Free     0               Art & Design  
4     4.3     100,000+  Free     0    Art & Design;Creativity

In [91]:
# Replace the '3.0M' in reviews col
Android_app[Android_app['Reviews'].str.contains('M') == True]
Android_app.drop(10472, inplace = True)

C:\Users\Admin\anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [92]:
# Reivews Column
Android_app['Reviews'] = Android_app['Reviews'].astype('int')

<ipython-input-92-28c304b533bd>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Android_app['Reviews'] = Android_app['Reviews'].astype('int')


In [93]:
Android_app.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10840 entries, 0 to 10840
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   App       10840 non-null  object 
 1   Category  10840 non-null  object 
 2   Reviews   10840 non-null  int32  
 3   Rating    9366 non-null   float64
 4   Installs  10840 non-null  object 
 5   Type      10839 non-null  object 
 6   Price     10840 non-null  object 
 7   Genres    10840 non-null  object 
dtypes: float64(1), int32(1), object(6)
memory usage: 719.8+ KB


In [94]:
# $
Android_app['Price'].str.contains('$').count()

10840

In [95]:
#########################

#Android_app['Price'] = Android_app['Price'].str.replace('$', "")
#Android_app['Price'].astype('float').dtype

In [96]:
# missing data
Android_app.isnull().sum()

App            0
Category       0
Reviews        0
Rating      1474
Installs       0
Type           1
Price          0
Genres         0
dtype: int64

No NaN value to remove

## Removing duplicated Apps

Have a quick scan throughout the labels, there are some apps that has been duplicated.

In [97]:
Android_app['App'].duplicated().sum()

1181

There are 1181 cases where apps occurs more than once. Examples of duplicate apps:

In [98]:
Android_app['Duplicated'] = Android_app['App'].duplicated()
# Android_app[Android_app['Duplicated'] == True]['App']
duplicate = Android_app.loc[Android_app['Duplicated'] == True]
duplicate.head()

<ipython-input-98-85fdfa651b2d>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Android_app['Duplicated'] = Android_app['App'].duplicated()


App  Category  Reviews  Rating     Installs  \
229  Quick PDF Scanner + OCR FREE  BUSINESS    80805     4.2   5,000,000+   
236                           Box  BUSINESS   159872     4.2  10,000,000+   
239            Google My Business  BUSINESS    70991     4.4   5,000,000+   
256           ZOOM Cloud Meetings  BUSINESS    31614     4.4  10,000,000+   
261     join.me - Simple Meetings  BUSINESS     6989     4.0   1,000,000+   

     Type Price    Genres  Duplicated  
229  Free     0  Business        True  
236  Free     0  Business        True  
239  Free     0  Business        True  
256  Free     0  Business        True  
261  Free     0  Business        True

In [99]:
Instagram = Android_app[Android_app['App'] == 'Instagram']

Remove the duplicated elements but Keep ones with the highest Reviews

In [100]:
Instagram['Reviews'].sort_values()

3909    66509917
2545    66577313
2611    66577313
2604    66577446
Name: Reviews, dtype: int32

In [101]:
duplicate.drop('Duplicated', axis =1, inplace = True)

C:\Users\Admin\anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [102]:
duplicate.sort_values('App').head()

App            Category  Reviews  Rating  \
1407         10 Best Foods for You  HEALTH_AND_FITNESS     2490     4.0   
2543    1800 Contacts - Lens Store             MEDICAL    23160     4.7   
2385    2017 EMRA Antibiotic Guide             MEDICAL       12     4.4   
1434  21-Day Meditation Experience  HEALTH_AND_FITNESS    11506     4.4   
5415       365Scores - Live Scores              SPORTS   666246     4.6   

         Installs  Type   Price            Genres  
1407     500,000+  Free       0  Health & Fitness  
2543   1,000,000+  Free       0           Medical  
2385       1,000+  Paid  $16.99           Medical  
1434     100,000+  Free       0  Health & Fitness  
5415  10,000,000+  Free       0            Sports

In [103]:
duplicate['App'].value_counts()+1

ROBLOX                                               9
CBS Sports App - Scores, News, Stats & Watch Live    8
Candy Crush Saga                                     7
8 Ball Pool                                          7
Duolingo: Learn Languages Free                       7
                                                    ..
Learn English with Wlingua                           2
Coursera: Online courses                             2
Alto's Adventure                                     2
United Airlines                                      2
10 Best Foods for You                                2
Name: App, Length: 798, dtype: int64

### Apply groupby.rank() method to retrieve the max reviews of each unique app 

In [104]:
Android_app['Rank'] = Android_app.groupby("App")['Reviews'].rank(method = 'first', ascending = False)

<ipython-input-104-d125a927c52e>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Android_app['Rank'] = Android_app.groupby("App")['Reviews'].rank(method = 'first', ascending = False)


In [105]:
Android_app[Android_app['App'].str.contains('Plants vs. Zombies FREE') == True]

App Category  Reviews  Rating      Installs  Type  \
1666  Plants vs. Zombies FREE     GAME  4066989     4.4  100,000,000+  Free   
1706  Plants vs. Zombies FREE     GAME  4066980     4.4  100,000,000+  Free   
1881  Plants vs. Zombies FREE     GAME  4067651     4.4  100,000,000+  Free   
1967  Plants vs. Zombies FREE     GAME  4067651     4.4  100,000,000+  Free   
4055  Plants vs. Zombies FREE   FAMILY  4064868     4.4  100,000,000+  Free   

     Price    Genres  Duplicated  Rank  
1666     0  Strategy       False   3.0  
1706     0  Strategy        True   4.0  
1881     0  Strategy        True   1.0  
1967     0  Strategy        True   2.0  
4055     0  Strategy        True   5.0

In [106]:
Android_app = Android_app[Android_app['Rank'] == 1]

In [107]:
Android_app.shape

(9659, 10)

After removing duplicates and used the max reviews of each apps, there are 9649 rows remaining

In [108]:
Android_app.drop(['Rank','Duplicated'], axis = 1, inplace= True)

In [109]:
Android_app.shape

(9659, 8)

## Removing Non-English Apps

In [110]:
Android_app.head(3)

App        Category  Reviews  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN      159   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN    87510   
3                              Sketch - Draw & Paint  ART_AND_DESIGN   215644   

   Rating     Installs  Type Price        Genres  
0     4.1      10,000+  Free     0  Art & Design  
2     4.7   5,000,000+  Free     0  Art & Design  
3     4.5  50,000,000+  Free     0  Art & Design

In [111]:
Android_app.iloc[4412][0]
Android_app.iloc[7940][0]

'لعبة تقدر تربح DZ'

Remove non-english classes

Explaination: All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it. ---> if there is a character in a string exceeds 127 so it returns not english

In [112]:
# Function that dectect non-english:

def is_english(string):
    
    for charater in string:
        if ord(charater) > 127:
            return False
        else:
            return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
print(ord('™'))
print(ord('😜'))

True
False
True
True
8482
128540


In [113]:
english_app = []

for app in list(Android_app['App']):
    if is_english(app):
        english_app.append(app)
        
print(f'# rows of english only app = {np.array(english_app).shape[0]}')
print(f'# rows of original data = {Android_app.shape[0]}')

# rows of english only app = 9623
# rows of original data = 9659


In [114]:
Android_app['App'] = Android_app[Android_app['App'].isin(english_app)]

# Free Apps

In [115]:
Android_app = Android_app[Android_app['Type'] == "Free"]

In [116]:
print(f'Free Android App = {Android_app.shape[0]}')

Free Android App = 8904


# EDA 

In [117]:
Android_app.head()

App        Category  Reviews  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN      159   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN    87510   
3                              Sketch - Draw & Paint  ART_AND_DESIGN   215644   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN      967   
5                         Paper flowers instructions  ART_AND_DESIGN      167   

   Rating     Installs  Type Price                   Genres  
0     4.1      10,000+  Free     0             Art & Design  
2     4.7   5,000,000+  Free     0             Art & Design  
3     4.5  50,000,000+  Free     0             Art & Design  
4     4.3     100,000+  Free     0  Art & Design;Creativity  
5     4.4      50,000+  Free     0             Art & Design

### By Category and Genre in GOOGLE PLAY STORE

In [118]:
# Category
Android_app['Category'].value_counts(normalize= True) * 100

FAMILY                 18.969003
GAME                    9.703504
TOOLS                   8.434412
BUSINESS                4.582210
LIFESTYLE               3.930818
PRODUCTIVITY            3.885894
FINANCE                 3.683738
MEDICAL                 3.515274
SPORTS                  3.380503
PERSONALIZATION         3.313118
COMMUNICATION           3.234501
HEALTH_AND_FITNESS      3.066038
PHOTOGRAPHY             2.942498
NEWS_AND_MAGAZINES      2.830189
SOCIAL                  2.650494
TRAVEL_AND_LOCAL        2.324798
SHOPPING                2.246181
BOOKS_AND_REFERENCE     2.178796
DATING                  1.853100
VIDEO_PLAYERS           1.796945
MAPS_AND_NAVIGATION     1.415094
FOOD_AND_DRINK          1.235400
EDUCATION               1.168014
ENTERTAINMENT           0.954627
LIBRARIES_AND_DEMO      0.932165
AUTO_AND_VEHICLES       0.920934
HOUSE_AND_HOME          0.819856
WEATHER                 0.797394
EVENTS                  0.707547
PARENTING               0.651393
ART_AND_DE

In Google Play Store, only 9.7% of total apps for gaming entertainment while Family category apps doubled at 18.97%. In top 10 most avaialbe app by category, there mostly apps for practical purposes (family, tools, business, lifestyle ... )  


In [119]:
# Genre
Android_app['Genres'].value_counts(normalize= True)[:20] * 100

Tools                8.423181
Entertainment        6.087152
Education            5.390836
Business             4.582210
Lifestyle            3.919587
Productivity         3.885894
Finance              3.683738
Medical              3.515274
Sports               3.447889
Personalization      3.313118
Communication        3.234501
Action               3.088500
Health & Fitness     3.066038
Photography          2.942498
News & Magazines     2.830189
Social               2.650494
Travel & Local       2.313567
Shopping             2.246181
Books & Reference    2.178796
Simulation           2.066487
Name: Genres, dtype: float64

The difference between Genres and Category columns is not crystal clear, but Genres column is more granular. % for practical purpose apps has been dominated for entertainment only.

Take a deeper look in Family Category

In [120]:
Family = Android_app[Android_app['Category'] == 'FAMILY']

In [121]:
Family.groupby(by= 'Genres').count().sort_values('App', ascending = False)['App'][:20]

Genres
Entertainment                  459
Education                      384
Simulation                     176
Casual                         134
Puzzle                          78
Role Playing                    72
Strategy                        65
Educational;Education           35
Educational                     33
Education;Education             25
Casual;Pretend Play             21
Racing;Action & Adventure       15
Puzzle;Brain Games              15
Casual;Action & Adventure       12
Entertainment;Music & Video     12
Arcade;Action & Adventure       11
Casual;Brain Games              11
Educational;Pretend Play         8
Action;Action & Adventure        8
Board;Brain Games                7
Name: App, dtype: int64

In [122]:
Family['Genres'].value_counts(normalize= True)[:10] * 100

Entertainment            27.353464
Education                22.912966
Simulation               10.479574
Casual                    7.933689
Puzzle                    4.618117
Role Playing              4.262877
Strategy                  3.907638
Educational;Education     2.072232
Educational               1.953819
Education;Education       1.480166
Name: Genres, dtype: float64

In FAMILY Category, 27% of the apps belongs to Entertaiment genre; Then, Educational purpose experiecne 22% of the Family apps ---> The Family Category was divided into 2 main genre: Entertainment with games, Casual, Puzzle or Role Playing; and Education 

In [123]:
Android_app.head()

App        Category  Reviews  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN      159   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN    87510   
3                              Sketch - Draw & Paint  ART_AND_DESIGN   215644   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN      967   
5                         Paper flowers instructions  ART_AND_DESIGN      167   

   Rating     Installs  Type Price                   Genres  
0     4.1      10,000+  Free     0             Art & Design  
2     4.7   5,000,000+  Free     0             Art & Design  
3     4.5  50,000,000+  Free     0             Art & Design  
4     4.3     100,000+  Free     0  Art & Design;Creativity  
5     4.4      50,000+  Free     0             Art & Design

Which Category has the highest rating on average on gg play store?

In [124]:
Android_app.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8904 entries, 0 to 10840
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   App       8872 non-null   object 
 1   Category  8904 non-null   object 
 2   Reviews   8904 non-null   int32  
 3   Rating    7594 non-null   float64
 4   Installs  8904 non-null   object 
 5   Type      8904 non-null   object 
 6   Price     8904 non-null   object 
 7   Genres    8904 non-null   object 
dtypes: float64(1), int32(1), object(6)
memory usage: 591.3+ KB


In [125]:
x = Android_app[Android_app['Category']=='ART_AND_DESIGN']
x[['Rating','Reviews']] = x[['Rating', 'Reviews']].dropna()

C:\Users\Admin\anaconda3\lib\site-packages\pandas\core\frame.py:3065: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]


In [126]:
x['W_Review'] = x['Reviews'] / x['Reviews'].sum() 

<ipython-input-126-a0e6a1544550>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  x['W_Review'] = x['Reviews'] / x['Reviews'].sum()


In [127]:
(x['W_Review']*x['Rating']).sum()

4.473816601355378

In [128]:
cat_list = Android_app['Category'].unique()
rating_avg = []
for cat in cat_list:
    
    x = Android_app[Android_app['Category']==cat]
    x[['Rating','Reviews']] = x[['Rating', 'Reviews']].dropna()
    
    x['W_Review'] = x['Reviews'] / x['Reviews'].sum() 
    avg=(x['W_Review']*x['Rating']).sum()
    rating_avg.append(round(avg,2))
print(rating_avg)
print('\n')
print(cat_list)

C:\Users\Admin\anaconda3\lib\site-packages\pandas\core\frame.py:3065: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]
<ipython-input-128-a2e00f4a8483>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  x['W_Review'] = x['Reviews'] / x['Reviews'].sum()
C:\Users\Admin\anaconda3\lib\site-packages\pandas\core\frame.py:3065: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the d

[4.47, 4.49, 4.34, 4.5, 4.32, 4.39, 4.28, 4.17, 4.56, 4.28, 4.19, 4.38, 4.38, 4.57, 4.42, 4.09, 4.32, 4.45, 4.4, 4.6, 4.27, 4.45, 4.43, 4.34, 4.36, 4.54, 4.51, 4.43, 4.62, 4.41, 4.4, 4.3, 4.4]


['ART_AND_DESIGN' 'AUTO_AND_VEHICLES' 'BEAUTY' 'BOOKS_AND_REFERENCE'
 'BUSINESS' 'COMICS' 'COMMUNICATION' 'DATING' 'EDUCATION' 'ENTERTAINMENT'
 'EVENTS' 'FINANCE' 'FOOD_AND_DRINK' 'HEALTH_AND_FITNESS' 'HOUSE_AND_HOME'
 'LIBRARIES_AND_DEMO' 'LIFESTYLE' 'GAME' 'FAMILY' 'MEDICAL' 'SOCIAL'
 'SHOPPING' 'PHOTOGRAPHY' 'SPORTS' 'TRAVEL_AND_LOCAL' 'TOOLS'
 'PERSONALIZATION' 'PRODUCTIVITY' 'PARENTING' 'WEATHER' 'VIDEO_PLAYERS'
 'NEWS_AND_MAGAZINES' 'MAPS_AND_NAVIGATION']


<ipython-input-128-a2e00f4a8483>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  x['W_Review'] = x['Reviews'] / x['Reviews'].sum()
C:\Users\Admin\anaconda3\lib\site-packages\pandas\core\frame.py:3065: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]
<ipython-input-128-a2e00f4a8483>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydat

In [129]:
dict_zip = zip(cat_list, rating_avg)
dict_cat_rating = dict(dict_zip)
df_cat_rating = pd.DataFrame(list(dict_cat_rating.items()), columns= ['Category', 'Average Ratings'])
df_cat_rating.sort_values(by= 'Average Ratings', ascending = False)

Category  Average Ratings
28            PARENTING             4.62
19              MEDICAL             4.60
13   HEALTH_AND_FITNESS             4.57
8             EDUCATION             4.56
25                TOOLS             4.54
26      PERSONALIZATION             4.51
3   BOOKS_AND_REFERENCE             4.50
1     AUTO_AND_VEHICLES             4.49
0        ART_AND_DESIGN             4.47
17                 GAME             4.45
21             SHOPPING             4.45
27         PRODUCTIVITY             4.43
22          PHOTOGRAPHY             4.43
14       HOUSE_AND_HOME             4.42
29              WEATHER             4.41
30        VIDEO_PLAYERS             4.40
18               FAMILY             4.40
32  MAPS_AND_NAVIGATION             4.40
5                COMICS             4.39
12       FOOD_AND_DRINK             4.38
11              FINANCE             4.38
24     TRAVEL_AND_LOCAL             4.36
23               SPORTS             4.34
2                BEAUTY             4.34
4              BUSINESS             4.32
16            LIFESTYLE             4.32
31   NEWS_AND_MAGAZINES             4.30
9         ENTERTAINMENT             4.28
6         COMMUNICATION             4.28
20               SOCIAL             4.27
10               EVENTS             4.19
7                DATING             4.17
15   LIBRARIES_AND_DEMO             4.09

Pretty high ratings based on reviews --> Parenting was the greatest, then medical, health and fitness. While Libraries and demo are the worst; 

### Installations

Which Category was most installed?

In [130]:
Android_app['Installs'].value_counts()

1,000,000+        1397
100,000+          1031
10,000,000+        935
10,000+            913
1,000+             750
100+               616
5,000,000+         607
500,000+           493
50,000+            429
5,000+             403
10+                315
500+               288
50,000,000+        204
100,000,000+       189
50+                170
5+                  70
1+                  46
500,000,000+        24
1,000,000,000+      20
0+                   4
Name: Installs, dtype: int64

In [131]:
Android_app['Installs'].nunique()

20

In [132]:
# Top installed apps
Android_app[Android_app['Installs'].isin(['1,000,000,000+', '500,000,000+'])].sort_values(by= 'Installs')

App             Category  \
152                                   Google Play Books  BOOKS_AND_REFERENCE   
5856                                  Google Play Games               FAMILY   
3687                            Google Play Movies & TV        VIDEO_PLAYERS   
3665                                            YouTube        VIDEO_PLAYERS   
3523                                       Google Drive         PRODUCTIVITY   
3234                                             Google                TOOLS   
3232                                 Google Street View     TRAVEL_AND_LOCAL   
3223                          Maps - Navigate & Explore     TRAVEL_AND_LOCAL   
2604                                          Instagram               SOCIAL   
2554                                            Google+               SOCIAL   
2544                                           Facebook               SOCIAL   
1872                                     Subway Surfers                 GAME   
2884                                      Google Photos          PHOTOGRAPHY   
9844                                        Google News   NEWS_AND_MAGAZINES   
336                                  WhatsApp Messenger        COMMUNICATION   
464                                            Hangouts        COMMUNICATION   
382            Messenger – Text and Video Chat for Free        COMMUNICATION   
451                                               Gmail        COMMUNICATION   
391                       Skype - free IM & video calls        COMMUNICATION   
411                        Google Chrome: Fast & Secure        COMMUNICATION   
3574                                        Cloud Print         PRODUCTIVITY   
371               Google Duo - High Quality Video Calls        COMMUNICATION   
3711                                          MX Player        VIDEO_PLAYERS   
3739                                            Twitter   NEWS_AND_MAGAZINES   
3819                       Flipboard: News For Our Time   NEWS_AND_MAGAZINES   
4005            Clean Master- Space Cleaner & Antivirus                TOOLS   
4676                                    Viber Messenger        COMMUNICATION   
5596                                     Samsung Health   HEALTH_AND_FITNESS   
3562                                    Google Calendar         PRODUCTIVITY   
383                       imo free video calls and chat        COMMUNICATION   
1885                                                Pou                 GAME   
1869                                   Candy Crush Saga                 GAME   
3265                       Gboard - the Google Keyboard                TOOLS   
3255                         SHAREit - Transfer & Share                TOOLS   
3235                                   Google Translate                TOOLS   
403                         LINE: Free Calls & Messages        COMMUNICATION   
7536  Security Master - Antivirus, VPN, AppLock, Boo...                TOOLS   
2610                                           Snapchat               SOCIAL   
420         UC Browser - Fast Download Private & Secure        COMMUNICATION   
2546                                      Facebook Lite               SOCIAL   
1920                                       Temple Run 2                 GAME   
1908                                     My Talking Tom                 GAME   
3473                                            Dropbox         PRODUCTIVITY   
3450                                     Microsoft Word         PRODUCTIVITY   

       Reviews  Rating        Installs  Type Price                   Genres  
152    1433233     3.9  1,000,000,000+  Free     0        Books & Reference  
5856   7168735     4.3  1,000,000,000+  Free     0            Entertainment  
3687    906384     3.7  1,000,000,000+  Free     0  Video Players & Editors  
3665  25655305     4.3  1,000,000,000+  Free     0  Video Players & Editors  
3523   2731211     4.4  1,000,000,000+  Free     0           

In [133]:
Android_app['Category'].value_counts()
Android_app['Category'].nunique()

33

In [134]:
pivot_cat_installs = pd.pivot_table(Android_app, index='Installs', columns='Category', aggfunc='count')
pivot_cat_installs

App                                               \
Category       ART_AND_DESIGN AUTO_AND_VEHICLES BEAUTY BOOKS_AND_REFERENCE   
Installs                                                                     
0+                        1.0               NaN    NaN                 NaN   
1+                        NaN               1.0    NaN                 NaN   
1,000+                    1.0               8.0    1.0                32.0   
1,000,000+                4.0              13.0    8.0                20.0   
1,000,000,000+            NaN               NaN    NaN                 1.0   
10+                       NaN               5.0    NaN                 4.0   
10,000+                  10.0               6.0   13.0                23.0   
10,000,000+               4.0               2.0    1.0                19.0   
100+                      1.0               5.0    5.0                 6.0   
100,000+                 18.0              23.0    8.0                20.0   
100,000,000+              NaN               NaN    NaN                 4.0   
5+                        NaN               2.0    NaN                 4.0   
5,000+                    4.0               2.0    3.0                14.0   
5,000,000+                3.0               3.0    1.0                 9.0   
50+                       NaN               NaN    1.0                 2.0   
50,000+                   6.0               4.0    5.0                12.0   
50,000,000+               1.0               NaN    NaN                 NaN   
500+                      NaN               3.0    1.0                 7.0   
500,000+                  4.0               5.0    6.0                16.0   
500,000,000+              NaN               NaN    NaN                 NaN   

                                                                             \
Category       BUSINESS COMICS COMMUNICATION DATING EDUCATION ENTERTAINMENT   
Installs                                                                      
0+                  1.0    NaN           NaN    NaN       NaN           NaN   
1+                  5.0    NaN           1.0    2.0       NaN           NaN   
1,000+             47.0    1.0          19.0   12.0       NaN           NaN   
1,000,000+         34.0   10.0          40.0   19.0      42.0          32.0   
1,000,000,000+      NaN    NaN           6.0    NaN       NaN           NaN   
10+                40.0    NaN          14.0   10.0       NaN           NaN   
10,000+            45.0   10.0          20.0   20.0       5.0           1.0   
10,000,000+        13.0    1.0          43.0    8.0       8.0          20.0   
100+               74.0    1.0          28.0   14.0       NaN           NaN   
100,000+           28.0   10.0          16.0   24.0      20.0           9.0   
100,000,000+        2.0    NaN          16.0    NaN       NaN           5.0   
5+                 17.0    NaN           2.0    1.0       NaN           NaN   
5,000+             17.0    4.0          16.0    9.0       NaN           NaN   
5,000,000+         15.0    4.0          22.0    6.0      12.0          11.0   
50+                24.0    1.0           5.0    5.0       NaN           NaN   
50,000+            11.0    6.0          10.0    5.0       6.0           1.0   
50,000,000+         5.0    NaN           7.0    NaN       NaN           4.0   
500+               22.0    NaN           8.0   12.0       NaN           NaN   
500,000+            8.0    5.0           9.0   18.0      11.0           1.0   
500,000,000+        NaN    NaN           5.0    NaN       NaN           NaN   

                ...            Type                                           \
Category        ... PERSONALIZATION PHOTOGRAPHY PRODUCTIVITY SHOPPING SOCIAL   
Installs        ...                                                            
0+              ...             NaN         NaN          NaN      NaN    1.0   
1+              ...             3.0         NaN          4.0      NaN    2.0   
1,000+       

Because Installs column include string data with "+" and "," character, which means the column did not represent accurately the installation for each app. So let's consider the number of installation for apps equal to the number on the web, for instance, 100+ means the app was installed 100 times

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.


In [135]:
# convert from string to int values
Installs = []
for x in list(Android_app['Installs']):
    y = x.replace('+', '').replace(',','')
    Installs.append(int(y))

# print(Installs)

In [136]:
# Replace string data in installs column by the list of integer

Android_app['Installs'] = Installs
Android_app['Installs'].dtype

dtype('int64')

In [137]:
aaa = Android_app[['Category', 'Installs', 'App']]
aaa.head()

Category  Installs                                                App
0  ART_AND_DESIGN     10000     Photo Editor & Candy Camera & Grid & ScrapBook
2  ART_AND_DESIGN   5000000  U Launcher Lite – FREE Live Cool Themes, Hide ...
3  ART_AND_DESIGN  50000000                              Sketch - Draw & Paint
4  ART_AND_DESIGN    100000              Pixel Draw - Number Art Coloring Book
5  ART_AND_DESIGN     50000                         Paper flowers instructions

In [138]:
pivot_cat_installs = pd.pivot_table(aaa, index='Category', columns='Installs', aggfunc='count')
pivot_cat_installs

App                                              \
Installs            0          1          5          10         50           
Category                                                                     
ART_AND_DESIGN             1.0        NaN        NaN        NaN        NaN   
AUTO_AND_VEHICLES          NaN        1.0        2.0        5.0        NaN   
BEAUTY                     NaN        NaN        NaN        NaN        1.0   
BOOKS_AND_REFERENCE        NaN        NaN        4.0        4.0        2.0   
BUSINESS                   1.0        5.0       17.0       40.0       24.0   
COMICS                     NaN        NaN        NaN        NaN        1.0   
COMMUNICATION              NaN        1.0        2.0       14.0        5.0   
DATING                     NaN        2.0        1.0       10.0        5.0   
EDUCATION                  NaN        NaN        NaN        NaN        NaN   
ENTERTAINMENT              NaN        NaN        NaN        NaN        NaN   
EVENTS                     NaN        NaN        NaN        6.0        NaN   
FAMILY                     NaN        6.0        5.0       46.0       24.0   
FINANCE                    NaN        2.0        NaN        3.0        6.0   
FOOD_AND_DRINK             NaN        1.0        2.0        4.0        4.0   
GAME                       NaN        NaN        2.0       14.0       10.0   
HEALTH_AND_FITNESS         NaN        2.0        2.0       14.0        9.0   
HOUSE_AND_HOME             NaN        1.0        NaN        1.0        1.0   
LIBRARIES_AND_DEMO         NaN        NaN        NaN        1.0        NaN   
LIFESTYLE                  NaN        4.0        1.0       17.0       11.0   
MAPS_AND_NAVIGATION        NaN        NaN        NaN        4.0        1.0   
MEDICAL                    NaN        4.0        8.0       25.0       21.0   
NEWS_AND_MAGAZINES         1.0        NaN        2.0       10.0        7.0   
PARENTING                  NaN        NaN        NaN        1.0        NaN   
PERSONALIZATION            NaN        3.0        1.0       13.0        3.0   
PHOTOGRAPHY                NaN        NaN        1.0        1.0        2.0   
PRODUCTIVITY               NaN        4.0        2.0       20.0       10.0   
SHOPPING                   NaN        NaN        1.0        8.0        NaN   
SOCIAL                     1.0        2.0        2.0        9.0        4.0   
SPORTS                     NaN        3.0        8.0       19.0        7.0   
TOOLS                      NaN        4.0        4.0       20.0       11.0   
TRAVEL_AND_LOCAL           NaN        1.0        3.0        2.0        1.0   
VIDEO_PLAYERS              NaN        NaN        NaN        2.0        NaN   
WEATHER                    NaN        NaN        NaN        2.0        NaN   

                                                                            \
Installs            100        500        1000       5000       10000        
Category                                                                     
ART_AND_DESIGN             1.0        NaN        1.0        4.0       10.0   
AUTO_AND_VEHICLES          5.0        3.0        8.0        2.0        6.0   
BEAUTY                     5.0        1.0        1.0        3.0       13.0   
BOOKS_AND_REFERENCE        6.0        7.0       32.0       14.0       23.0   
BUSINESS                  74.0       22.0       47.0       17.0       45.0   
COMICS                     1.0        NaN        1.0        4.0       10.0   
COMMUNICATION             28.0        8.0       19.0       16.0       20.0   
DATING                    14.0       12.0       12.0        9.0       20.0   
EDUCATION                  NaN        NaN        NaN        NaN        5.0   
ENTERTAINMENT              NaN        NaN        NaN        NaN        1.0   
EVENTS                    11.0        6.0        9.0        4.0        4.0   
FAMILY                    95.0       63.0      189.0       93.0      192.0   
FINANCE                   23.0       11.0       

Have a quick look into the Pivot matrix:
- In 20 apps which has more than 1B downloads, communication/social network included 6 apps, such as Facebook, Instagramm, Twitter, etc; The other spreaded from Books and Referance category to SAport, Photography, Travel, Tools, etc
- In the second group of more than 500M installs, Communication, Productivity, Game and Tools dominated with approximately 20 out of 25 apps with the similarity of installation

In [139]:
Installs_cat = pd.DataFrame(Android_app.groupby(by= 'Category')['Installs'].sum()).sort_values(by= 'Installs', ascending = False)

In [140]:
Installs_cat['Number_Apps'] = Android_app['Category'].value_counts()
Installs_cat['Install per App'] = Installs_cat['Installs'] / Installs_cat['Number_Apps']

In [141]:
Cat_df = Installs_cat.merge(df_cat_rating, how= 'left', on= 'Category')
Cat_df

Category     Installs  Number_Apps  Install per App  \
0                  GAME  13436924450          864     1.555200e+07   
1         COMMUNICATION  11036916201          288     3.832263e+07   
2                 TOOLS   8101044474          751     1.078701e+07   
3                FAMILY   6200391691         1689     3.671043e+06   
4          PRODUCTIVITY   5791679314          346     1.673896e+07   
5                SOCIAL   5487861902          236     2.325365e+07   
6           PHOTOGRAPHY   4656268915          262     1.777202e+07   
7         VIDEO_PLAYERS   3931831720          160     2.457395e+07   
8      TRAVEL_AND_LOCAL   2894704086          207     1.398408e+07   
9    NEWS_AND_MAGAZINES   2369212260          252     9.401636e+06   
10  BOOKS_AND_REFERENCE   1665946260          194     8.587352e+06   
11      PERSONALIZATION   1529235988          295     5.183851e+06   
12             SHOPPING   1400338685          200     7.001693e+06   
13   HEALTH_AND_FITNESS   1143548402          273     4.188822e+06   
14               SPORTS   1095230683          301     3.638640e+06   
15        ENTERTAINMENT    989460000           85     1.164071e+07   
16             BUSINESS    696952090          408     1.708216e+06   
17  MAPS_AND_NAVIGATION    503160790          126     3.993340e+06   
18            LIFESTYLE    502644429          350     1.436127e+06   
19              FINANCE    455163132          328     1.387692e+06   
20              WEATHER    360288520           71     5.074486e+06   
21       FOOD_AND_DRINK    211738751          110     1.924898e+06   
22            EDUCATION    189850000          104     1.825481e+06   
23               DATING    140914757          165     8.540288e+05   
24       ART_AND_DESIGN    113222100           58     1.952105e+06   
25       HOUSE_AND_HOME     97202461           73     1.331541e+06   
26    AUTO_AND_VEHICLES     53080061           82     6.473178e+05   
27   LIBRARIES_AND_DEMO     52995810           83     6.385037e+05   
28               COMICS     44981150           56     8.032348e+05   
29              MEDICAL     37732344          313     1.205506e+05   
30            PARENTING     31471010           58     5.426036e+05   
31               BEAUTY     27197050           53     5.131519e+05   
32               EVENTS     15973160           63     2.535422e+05   

    Average Ratings  
0              4.45  
1              4.28  
2              4.54  
3              4.40  
4              4.43  
5              4.27  
6              4.43  
7              4.40  
8              4.36  
9              4.30  
10             4.50  
11             4.51  
12             4.45  
13             4.57  
14             4.34  
15             4.28  
16             4.32  
17             4.40  
18             4.32  
19             4.38  
20             4.41  
21             4.38  
22             4.56  
23             4.17  
24             4.47  
25             4.42  
26             4.49  
27             4.09  
28             4.39  
29             4.60  
30             4.62  
31             4.34  
32             4.19

In [152]:
# change format from scienctific notion to normal format
Cat_df['Install per App'] = Cat_df['Install per App'].apply(lambda x: '%.f' %x)

In [153]:
Cat_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 33 entries, 0 to 32
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Category         33 non-null     object 
 1   Installs         33 non-null     int64  
 2   Number_Apps      33 non-null     int64  
 3   Install per App  33 non-null     object 
 4   Average Ratings  33 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 1.5+ KB


In [154]:
# the Install per app data was string -> need convert to float
Cat_df['Install per App'] = Cat_df['Install per App'].astype('float')

In [155]:
# Inisght the average install per app
Cat_df.sort_values(by= 'Install per App', ascending = False)

Category     Installs  Number_Apps  Install per App  \
1         COMMUNICATION  11036916201          288       38322626.0   
7         VIDEO_PLAYERS   3931831720          160       24573948.0   
5                SOCIAL   5487861902          236       23253652.0   
6           PHOTOGRAPHY   4656268915          262       17772019.0   
4          PRODUCTIVITY   5791679314          346       16738958.0   
0                  GAME  13436924450          864       15551996.0   
8      TRAVEL_AND_LOCAL   2894704086          207       13984078.0   
15        ENTERTAINMENT    989460000           85       11640706.0   
2                 TOOLS   8101044474          751       10787010.0   
9    NEWS_AND_MAGAZINES   2369212260          252        9401636.0   
10  BOOKS_AND_REFERENCE   1665946260          194        8587352.0   
12             SHOPPING   1400338685          200        7001693.0   
11      PERSONALIZATION   1529235988          295        5183851.0   
20              WEATHER    360288520           71        5074486.0   
13   HEALTH_AND_FITNESS   1143548402          273        4188822.0   
17  MAPS_AND_NAVIGATION    503160790          126        3993340.0   
3                FAMILY   6200391691         1689        3671043.0   
14               SPORTS   1095230683          301        3638640.0   
24       ART_AND_DESIGN    113222100           58        1952105.0   
21       FOOD_AND_DRINK    211738751          110        1924898.0   
22            EDUCATION    189850000          104        1825481.0   
16             BUSINESS    696952090          408        1708216.0   
18            LIFESTYLE    502644429          350        1436127.0   
19              FINANCE    455163132          328        1387692.0   
25       HOUSE_AND_HOME     97202461           73        1331541.0   
23               DATING    140914757          165         854029.0   
28               COMICS     44981150           56         803235.0   
26    AUTO_AND_VEHICLES     53080061           82         647318.0   
27   LIBRARIES_AND_DEMO     52995810           83         638504.0   
30            PARENTING     31471010           58         542604.0   
31               BEAUTY     27197050           53         513152.0   
32               EVENTS     15973160           63         253542.0   
29              MEDICAL     37732344          313         120551.0   

    Average Ratings  
1              4.28  
7              4.40  
5              4.27  
6              4.43  
4              4.43  
0              4.45  
8              4.36  
15             4.28  
2              4.54  
9              4.30  
10             4.50  
12             4.45  
11             4.51  
20             4.41  
13             4.57  
17             4.40  
3              4.40  
14             4.34  
24             4.47  
21             4.38  
22             4.56  
16             4.32  
18             4.32  
19             4.38  
25             4.42  
23             4.17  
28             4.39  
26             4.49  
27             4.09  
30             4.62  
31             4.34  
32             4.19  
29             4.60

### Scanning some top Installs per App

In [176]:
# COMMUNICATION
Android_app[Android_app['Category'] == 'COMMUNICATION'][['App','Installs']].sort_values(by = 'Installs', ascending = False)[:10]

App    Installs
336                            WhatsApp Messenger  1000000000
391                 Skype - free IM & video calls  1000000000
411                  Google Chrome: Fast & Secure  1000000000
382      Messenger – Text and Video Chat for Free  1000000000
451                                         Gmail  1000000000
464                                      Hangouts  1000000000
403                   LINE: Free Calls & Messages   500000000
383                 imo free video calls and chat   500000000
4676                              Viber Messenger   500000000
420   UC Browser - Fast Download Private & Secure   500000000

In top 20 of communication Apps, they are all ubiquitous Message Apps or Browsers, so the competition advantage was belonging to the giant techs, they all dominated the installations on google store. 

In [165]:
# SOCIAL
Android_app[Android_app['Category'] == 'SOCIAL'][['App','Installs']].sort_values(by = 'Installs', ascending = False)[:10]

App    Installs
2544                        Facebook  1000000000
2554                         Google+  1000000000
2604                       Instagram  1000000000
2610                        Snapchat   500000000
2546                   Facebook Lite   500000000
3945  Tik Tok - including musical.ly   100000000
2592    Tango - Live Video Broadcast   100000000
6373                              VK   100000000
2552                       Pinterest   100000000
3951         BIGO LIVE - Live Stream   100000000

The similar as true for Social apps

In [166]:
# VIDEO_PLAYERS
Android_app[Android_app['Category'] == 'VIDEO_PLAYERS'][['App','Installs']].sort_values(by = 'Installs', ascending = False)[:10]

App    Installs
3665                                             YouTube  1000000000
3687                             Google Play Movies & TV  1000000000
3711                                           MX Player   500000000
4696   VideoShow-Video Editor, Video Maker, Beauty Ca...   100000000
4688              VivaVideo - Video Editor & Photo Movie   100000000
3672                                    Motorola Gallery   100000000
3675                                     VLC for Android   100000000
4032                                            Dubsmash   100000000
10647                                  Motorola FM Radio   100000000
5612                                           Ringdroid    50000000

These top categories seems to be the fight between giant developers because of the profitability and the markets were segmented enough for them to dominate other developers, and fighting against these giants is quite impossible. 

The Genres like Game, Family and Tools dominate the google store by there apps uploaded, means fairly popular as well. However, the problem with Family and Tools did not attract a lot of users; While gaming sections were a bit satuarted: easy to entry but to gain profits is not a simple question.

In my opinion, what we should look for is a market with average competitition but still gains good enough number of users. 

BOOKS_AND_REFERENCE is quite interesting to explore in more depth, with the average number of installation was 8.4M (top10 in avg installaiton) and the number of apps on Google Store was nearly 200 apps.

Let's have a glance on BOOKS_AND_REFERENCE

In [177]:
Book_Ref_df = Android_app[Android_app['Category'] == 'BOOKS_AND_REFERENCE'][['App', 'Installs']]
Book_Ref_df.sort_values(by= 'Installs', ascending = False)[:20]

App    Installs
152                                   Google Play Books  1000000000
5651                            Audiobooks from Audible   100000000
4715                               Wattpad 📖 Free Books   100000000
3941                                              Bible   100000000
4083                                      Amazon Kindle   100000000
6497                      NOOK: Read eBooks & Magazines    10000000
5324                          Al'Quran Bahasa Indonesia    10000000
179                                        Moon+ Reader    10000000
8293                                         Dictionary    10000000
5345                                  Quran for Android    10000000
5323                                 Al Quran Indonesia    10000000
5319                                    Al-Quran (Free)    10000000
4100                                 Aldiko Book Reader    10000000
9643                           English Hindi Dictionary    10000000
173                                            HTC Help    10000000
9642                Oxford Dictionary of English : Free    10000000
6296                       English Dictionary - Offline    10000000
9625                                         JW Library    10000000
6290  Dictionary.com: Find Definitions for English W...    10000000
149                      FBReader: Favorite Book Reader    10000000

A lot of kind of book apps from Dictionary, PDF Assistant, Noting, E-book, E-Libraries, Religious, etc. Howerver, it looks like there are a few popular apps such as Google Play Book (1B+), Bibie (100M+), Amazon Kindle (100M+), Wattpad(100M+), Audible (100M+)... alongsides a many apps with more than 10M+ apps. So, this market is sitll potential for developers.

On the other hand, the apps here was quite similar in functions: software for processing and reading e-book or libraries, e-bookstores and dictionaries were dominant, so being different by including a raw but convenient function might be the way to get profitable instead of create an app with alike function that'd already existed.